src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -1681,9 +1681,26 @@ internal bool ShutdownLoop
                 }
             }
 
+            /// <summary>
+            /// Call WaitForCommEvent (which is a thread function for a background thread)
+            /// within an exception handler, so that unhandled exceptions in WaitForCommEvent
+            /// don't cause process termination
+            /// </summary>
+            internal void SafelyWaitForCommEvent()
+            {
+                try
+                {
+                    WaitForCommEvent();
+                }
+                catch(Exception ex)
```

Comment:
We should create an issue around improving thread usage that details what we're suffering from and what we'd be looking to improve. Tests need an issue as well- not surfacing exceptions makes tracking test failures a pain.

Putting comments here describing the ideal of surfacing exceptions to the main thread and referencing the issue would be great.

Assuming we open an issue I'm ok with incremental forward steps here-- as long as we're tracking towards longer term goals and not impeding future progress with our intermediate steps.